lib/scala/pkg/src/main/scala/org/enso/pkg/ComponentGroup.scala

```diff
@@ -0,0 +1,316 @@
+      val icon  = extendedComponentGroup.icon.map(Fields.Icon -> _.asJson)
+      val exports = Option.unless(extendedComponentGroup.exports.isEmpty)(
+        Fields.Exports -> extendedComponentGroup.exports.asJson
+      )
+      Json.obj(
+        (Fields.Module -> extendedComponentGroup.module.asJson) +:
+        (color.toSeq ++ icon.toSeq ++ exports.toSeq): _*
+      )
+  }
+
+  /** [[Decoder]] instance for the [[ExtendedComponentGroup]]. */
+  implicit val decoder: Decoder[ExtendedComponentGroup] = { json =>
+    for {
+      reference <- ConfigCodecs
+        .getFromObject[ModuleReference](
+          "extended component group reference",
+          Fields.Module,
+          json
+        )
+      color   <- json.get[Option[String]](Fields.Color)
+      icon    <- json.get[Option[String]](Fields.Icon)
+      exports <- json.getOrElse[List[Component]](Fields.Exports)(List())
+    } yield ExtendedComponentGroup(reference, color, icon, exports)
+  }
+}
+
+/** A single component of a component group.
+  *
+  * @param name the component name
+  * @param shortcut the component shortcut
+  */
+case class Component(name: String, shortcut: Option[Shortcut])
+object Component {
+
+  object Fields {
+    val Name     = "name"
+    val Shortcut = "shortcut"
+  }
+
+  /** [[Encoder]] instance for the [[Component]]. */
+  implicit val encoder: Encoder[Component] = { component =>
+    component.shortcut match {
+      case Some(shortcut) =>
+        Json.obj(
+          Fields.Name     -> component.name.asJson,
+          Fields.Shortcut -> shortcut.asJson
+        )
+      case None =>
+        component.name.asJson
+    }
+  }
+
+  /** [[Decoder]] instance for the [[Component]]. */
+  implicit val decoder: Decoder[Component] = { json =>
+    json.as[String] match {
+      case Right(name) =>
+        Right(Component(name, None))
+      case Left(_) =>
+        for {
+          name <- ConfigCodecs
+            .getFromObject[String]("component name", Fields.Name, json)
+          shortcut <- json.getOrElse[Option[Shortcut]](Fields.Shortcut)(None)
+        } yield Component(name, shortcut)
+    }
+  }
+}
+
+/** The shortcut reference to the component.
+  *
+  * @param key the shortcut key combination
+  */
+case class Shortcut(key: String)
+object Shortcut {
+
+  /** [[Encoder]] instance for the [[Shortcut]]. */
+  implicit val encoder: Encoder[Shortcut] = { shortcut =>
+    shortcut.key.asJson
+  }
+
+  /** [[Decoder]] instance for the [[Shortcut]]. */
+  implicit val decoder: Decoder[Shortcut] = { json =>
+    ConfigCodecs.getScalar("shortcut", json).map(Shortcut(_))
+  }
+}
+
+/** The reference to a module.
+  *
+  * @param libraryName the qualified name of a library where the module is defined
+  * @param moduleName the module name
+  */
+case class ModuleReference(
+  libraryName: LibraryName,
+  moduleName: Option[ModuleName]
+)
+object ModuleReference {
+
+  private val ModuleSeparator: Char = '.'
```

Comment:
This is also defined in `LibraryName.separator`, although currently private. It may be worth making it public and using the same constant in all places - it is unlikely we are going to change it, but seems a bit unnecessary to define the same constant twice.